### DISCUSSION TITLE
UseBoundstore<...> of create is deprecated

### DISCUSSION THREAD

**Discussion by garciabruno** [pmndrs/zustand#2570] [Q&A] [answered]
I have the following store definition that I was using in zustand 3.x, and when migrated to 4.x.
I get a Deprecation Warning when using `create(store)`.

My setup is similar to this:

```typescript
interface TState {
  myState: string;
};

interface UpdateMyState {
  type: "UPDATE_MY_STATE",
  payload: {
    myState: string;
  }
}

type TActions = UpdateMyState;

type TDispatch = {
  dispatch: (a: TActions) => TActions;
};

type TStateWithDispatch = TState & TDispatch;

[...]

export const store = createStore(
  persist(
    subscribeWithSelector(redux(reducer, initialState as TStateWithDispatch)),
    {
      name: "myApp",
      getStorage: () => AsyncStorage,
    }
  )
);

export const useTrackedSelector = createTrackedSelector(create(store));
```

The deprecation warning I get is the following:

```
The signature (store: WithRedux>>, StorePersist>, Write<...>>>, StoreSubscribeWithSelector<...>>, TActions>): UseBoundStore<...> of create is deprecated.
```

I've been trying for several hours to find the way to solve this deprecation without any luck.
Is there anyone that might point me in the right direction?

Any help is appreciated.

**Comment by dbritto-dev** [comment]
@garciabruno you are wrong, should be something like this: `createStore<Store>()(...)` when you use typescript and middlewares

**Comment by garciabruno** [comment]
By the `Store` type, in my case, you refer to `TStateWithDispatch`, right?

**Comment by garciabruno** [comment, marked as answer]
I solved it by using `create` instead of `createStore`, and passing that result to `createTrackedSelector`.

```typescript
const store = create(
  persist(
    subscribeWithSelector(redux(reducer, initialState as TStateWithDispatch)),
    {
      name: "myApp",
      storage: createJSONStorage(() => AsyncStorage),
    }
  )
);

const useTrackedSelector = createTrackedSelector(store);
```
